### PR TITLE
Efile state machine takes data source that will flip between tax returns and statefile intakes

### PR DESCRIPTION
**Pull request by embarnard**

### CODE REVIEW

File: app/models/efile_submission.rb [hunk 1]
```diff
@@ -141,6 +141,14 @@ def previously_transmitted_submission
     previous_submission if previous_submission.last_transition_to("transmitted").present?
   end
 
+  def source_record
+    if ["StateFileNyIntake", "StateFileAzIntake"].include? data_source_type
```
Comment (bengolder) [inline app/models/efile_submission.rb:145]: we could potentially improve this line by saying
```ruby
is_for_state_filing? ? data_source : tax_return
```

Comment (bengolder) [reply app/models/efile_submission.rb:145]: I hadn't realized that there were already methods to tell if it was for state vs federal filing

Comment (bengolder) [reply app/models/efile_submission.rb:145]: but this is definitely not blocking!

**Review by bengolder — APPROVED** [review]

File: app/models/efile_submission.rb (hunk 1 above)
Comment (embarnard) [reply app/models/efile_submission.rb]: this is a good change! added it

File: app/state_machines/efile_submission_state_machine.rb [hunk 2]
```diff
@@ -151,20 +150,20 @@ class EfileSubmissionStateMachine
   end
 
   after_transition(to: :investigating) do |submission|
-    submission.tax_return.transition_to(:file_hold)
+    submission.source_record.transition_to(:file_hold)
   end
```
Comment (bengolder) [inline app/state_machines/efile_submission_state_machine.rb:154]:
```suggestion
    submission.source_record.transition_to(:file_hold) if submission.is_for_federal_filing?
```

File: app/state_machines/efile_submission_state_machine.rb [hunk 3]
```diff
@@ -151,20 +150,20 @@ class EfileSubmissionStateMachine
...
 
   after_transition(to: :waiting) do |submission|
-    submission.tax_return.transition_to(:file_hold)
+    submission.source_record.transition_to(:file_hold)
   end
```
Comment (bengolder) [inline app/state_machines/efile_submission_state_machine.rb:158]:
```suggestion
    submission.source_record.transition_to(:file_hold) if submission.is_for_federal_filing?
```

File: app/state_machines/efile_submission_state_machine.rb [hunk 4]
```diff
@@ -151,20 +150,20 @@ class EfileSubmissionStateMachine
...
   after_transition(to: :resubmitted) do |submission, transition|
-    @new_submission = submission.tax_return.efile_submissions.create
+    @new_submission = submission.source_record.efile_submissions.create
     @new_submission.transition_to!(:preparing, previous_submission_id: submission.id, initiated_by_id: transition.metadata["initiated_by_id"])
   end
 
   after_transition(to: :cancelled) do |submission|
-    submission.tax_return.transition_to(:file_not_filing)
+    submission.source_record.transition_to(:file_not_filing)
   end
```
Comment (bengolder) [inline app/state_machines/efile_submission_state_machine.rb:167]:
```suggestion
    submission.source_record.transition_to(:file_not_filing) if submission.is_for_federal_filing?
```

File: app/state_machines/efile_submission_state_machine.rb [hunk 5]
```diff
@@ -110,6 +108,7 @@ class EfileSubmissionStateMachine
 
     if transition.efile_errors.any?
       if transition.efile_errors.any?(&:expose) && submission.is_for_federal_filing?
+        # this will not run for state file
```
Comment (bengolder) [inline app/state_machines/efile_submission_state_machine.rb:111]:
```suggestion
```

File: app/services/efile/poll_for_acknowledgments_service.rb [hunk 6]
```diff
@@ -47,6 +47,7 @@ def self.transmitted_submission_ids
     def self.transmitted_state_submission_ids
       transmitted_submissions = EfileSubmission.in_state(:transmitted)
       state_submissions = transmitted_submissions.for_state_filing
+      transmitted_submissions.touch_all(:last_checked_for_ack_at)
```
Comment (bengolder) [inline app/services/efile/poll_for_acknowledgments_service.rb:50]:
```suggestion
      state_submissions.touch_all(:last_checked_for_ack_at)
```

**Review by bengolder — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**